src/MainWindow.cpp

```diff
@@ -1531,3 +1549,48 @@ QMessageBox::StandardButton MainWindow::openNewFileQuestionDialog(const QString&
           QMessageBox::Yes | QMessageBox::No, QMessageBox::No
     );
 }
+
+// TODO: Reuse code between generateBashRemediationRole and generateAnsibleRemediationRole
+void MainWindow::generateBashRemediationRole()
+{
+    const QString filename = QFileDialog::getSaveFileName(this,
+        QObject::tr("Save as bash script"),
+        QObject::tr("remediate-profile.sh"),
+        QObject::tr("bash scripts (*.sh)"), 0
+#ifndef SCAP_WORKBENCH_USE_NATIVE_FILE_DIALOGS
+        , QFileDialog::DontUseNativeDialog
+#endif
+    );
+
+    if (filename.isEmpty())
+        return;
+
+    QFile output_file(filename);
+    output_file.open(QIODevice::WriteOnly);
+    struct xccdf_session * session = mScanningSession->getXCCDFSession();
+    struct xccdf_policy *policy = xccdf_session_get_xccdf_policy(session);
+    xccdf_policy_generate_fix(policy, NULL, "urn:xccdf:fix:script:sh", output_file.handle());
+    output_file.close();
+}
+
+void MainWindow::generateAnsibleRemediationRole()
+{
+    const QString filename = QFileDialog::getSaveFileName(this,
+        QObject::tr("Save as Ansible playbook"),
+        QObject::tr("remediate-profile.yml"),
+        QObject::tr("ansible playbooks (*.yml)"), 0
+#ifndef SCAP_WORKBENCH_USE_NATIVE_FILE_DIALOGS
+        , QFileDialog::DontUseNativeDialog
+#endif
+    );
+
+    if (filename.isEmpty())
+        return;
+
+    QFile output_file(filename);
+    output_file.open(QIODevice::WriteOnly);
+    struct xccdf_session * session = mScanningSession->getXCCDFSession();
+    struct xccdf_policy *policy = xccdf_session_get_xccdf_policy(session);
```

Comment:
code style for the above 2 lines. it sucks but code style in S-W is different from openscap